### ISSUE TITLE
[Feature]: Parallelize Flip 7 Evaluation

### ISSUE THREAD

**Issue by donaldng05** [donaldng05/flip7#19]
### Problem or research question

Full experiment matrix budgets take hours (~5–7 min per training run × 12–20 runs per matrix), dominated by sequential game simulation rather than Python vs. C++ execution speed.

While tournament round-robins already have a proven multiprocessing pattern in `evaluation/follow_up.py:run_paired_round_robin`, the standard rotated evaluation functions (`evaluation/phase6.py:run_rotated_matchups`, `evaluation/tournament.py:run_matchup`, and `evaluation/stability.py:run_paired_rotated_evaluation`) remain strictly sequential. These functions evaluate independent seeded games across baseline and held-out rosters, making evaluation an embarrassingly parallel workload and the highest-leverage first step to reducing experimental cycle time.

### Proposed change

Parallelize evaluation workloads using standard library `concurrent.futures.ProcessPoolExecutor` following the pattern established in `run_paired_round_robin`:

### 1. Target Functions
- `flip7.evaluation.phase6:run_rotated_matchups`: Parallelize game execution while preserving deterministic `MatchupMetrics` aggregation.
- `flip7.evaluation.tournament:run_matchup`: Support multi-worker execution over scheduled games.
- `flip7.evaluation.stability:run_paired_rotated_evaluation`: Parallelize over paired games on identical seed blocks (ensuring a pair is never split across disparate seeds).

### 2. Process Boundary & Concurrency Discipline
- Pre-generate deterministic game schedules upfront.
- Policies cross the process boundary as checkpoint paths (via serializable participant metadata), not live PyTorch objects or closures.
- Call `torch.set_num_threads(1)` in worker initializers to eliminate intra-op thread contention.
- Collect results in schedule order before computing metrics or summary dictionaries to ensure output determinism.

### 3. Configuration & CLI Plumbing
- Add a `workers: int = 1` parameter to the target functions and thread it through `experiment/eval.py:run_standard_evaluations`.
- Add `evaluation.workers` (default 1) to experiment YAML configs alongside `tournament.workers`.
- Wire `--workers` CLI arguments through `scripts/run_phase6.py`, `scripts/run_phase7.py`, `scripts/run_phase7_follow_up.py`, and `scripts/run_phase7_follow_up_stability.py`.
- Preserve default `workers=1` across all entry points for bit-identical reproducibility with historical baselines.

### Validation plan

- [ ] **Quality Gates:** `scripts/check.ps1` / `scripts/check.sh` passes fully:
  - Ruff format (`ruff format --check .`)
  - Ruff check (`ruff check .`)
  - Pyright strict
  - Pytest with coverage >= 80%
- [ ] **Worker Invariance Unit Tests:** Assert `workers=1` vs. `workers=4` produce bit-identical `MatchupMetrics`, `PairedEvaluation`, and summary dictionaries on fixed seeds across all parallelized entry points.
- [ ] **Fast-Loop Guardrail Oracle:**
  - Execute:
    ```bash
    uv run python scripts/run_phase6.py --config artifacts/guardrail/guardrail.yaml --output-root artifacts/guardrail/run-eval
    ```
  - Confirm:
    - Seed 7 win share falls within [50.3%, 64.7%] (spread <= 6.7pp + 5pp).
    - Seed 17 win share falls within [63.0%, 76.4%] (spread <= 10.8pp + 5pp).
    - Spread does not grow by > 5pp.
- [ ] **Wall-Clock Report:** Measure and record before/after wall time demonstrating near-linear scaling on multi-core CPU.
